### PR TITLE
release v4.1.0

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

Let's tag v4.1.0.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-4.1.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Release notes for v4.1.0
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-4.1.md:1]: Looks like the README should also be updated

Comment (pohly) [reply CHANGELOG/CHANGELOG-4.1.md:1]: The top-level one? Yes, that's severely outdated.

I've pushed a commit which updates it and the mock/README.md.

**Pull request merged**